CONTRIBUTING.md

```diff
@@ -96,6 +96,120 @@ Every commit in your pull request needs to be able to build and pass the CI test
 
 If the pull request closes an issue please note it as: `"Fixes #NNN"`.
 
+## Unit Testing Guidelines
+
+### Unit Test Checks
+
+* The "testify" package should not be used.
+* The "cmp" package is allowed.
+* Unit tests in Go should follow the guidelines in this tutorial: https://go.dev/doc/tutorial/add-a-test
+  * In particular, the test error should be in the form `Function(...) = ...; want ...`.
+
+For example:
+
+```
+if msg != "" || err == nil {
+	t.Fatalf(`Hello("") = %q, %v, want "", error`, msg, err)
+}
+```
+
+### Mocking Dependencies
+
+* Special mocking packages should not be used.
+* Prefer to test the real thing where possible.
+* Mocking is sometimes necessary. For example:
+   * Operations as root.
+   * Interacting with special hardware (ex: USB, SPI, PCIe)
+   * Modifying machine state (ex: reboot, kexec)
+   * Tests which would otherwise be flaky (ex: `time.Sleep`, networking)
+* Consider writing an integration test for the above.
```

Comment:
This needs some explaining. 
* integration tests let you run the command under qemu, which lets you test operations with, e.g., virtual hardware. Consider writing an integration test if the program can not be easily run directly.

Reply:
Fixed!